xjc-maven-plugin/src/main/java/com/github/davidmoten/xjc/XjcMojo.java

```diff
@@ -292,16 +284,26 @@ private static String string(Artifact a) {
         return a.getGroupId() + ":" + a.getArtifactId() + ":" + a.getVersion() + ":" + a.getScope() + ":" + a.getType();
     }
 
-    private void ensureDestinationDirectoryExists() {
+    private File createOutputDirectoryIfSpecifiedOrDefault() {
         for (int i = 0; i < arguments.size(); i++) {
-            if (arguments.get(i).trim().equals("-d") && i < arguments.size() - 1) {
-                outputDir = new File(arguments.get(i + 1));
+            if (isDirParamSpecifiedAndNotEmpty(arguments, i)) {
+                File outputDir = new File(arguments.get(i + 1));
                 if (!outputDir.exists()) {
                     getLog().info("destination directory (-d option) specified and does not exist, creating: " + outputDir);
                     outputDir.mkdirs();
+                    return outputDir;
                 }
             }
         }
+
+        final File outputDir = new File(mavenProject.getBuild().getDirectory());
```

Comment:
I would expect this to be `return new File(".");`. If otherwise then the `ProcessExecutor` needs to have its current working directory set. I'm not keen on that and prefer `new File(".")` so there are no surprises as to where the output of the xjc command ends up.

Reply:
All right, will fix it, however keep in mind that generated sources will then be generated in project root(which possibly clashes with VCS settings in IDE)

Reply:
Yep I'm alright with that. I can add a reminder in the README.md about what happens if you leave out the `-d` parameter. An alternative is that we add a `-d` parameter ourselves if one doesn't exist but I'd prefer not to muck with the call of `xjc` at all so the user has all the control.